virtool/config.py

```diff
@@ -136,8 +139,49 @@ def cli(ctx, data_path, db_connection_string, db_name, dev, force_version, no_se
     help="Start with automatic fetching disabled",
     is_flag=True
 )
+@click.option(
+    "--use-b2c",
+    help="Use OIDC for authentication",
+    is_flag=True
+)
+@click.option(
+    "--b2c-tenant",
+    help="Azure AD B2C tenant name",
+    type=str,
+    envvar="B2C_TENANT"
+)
+@click.option(
+    "--b2c-client-id",
+    type=str,
+    envvar="OIDC_CLIENT_ID",
+    help="Keycloak client ID, required for --use-oidc"
+)
+@click.option(
+    "--b2c-client-secret",
+    type=str,
+    envvar="OIDC_CLIENT_SECRET",
+    help="Keycloak client secret, required for --use-oidc"
+)
+@click.option(
+    "--b2c-user-flow",
+    type=str,
+    envvar="B2C_USER_FLOW"
+)
 @click.pass_context
-def start_server(ctx, host, port, no_check_db, no_check_files, no_client, no_fetching):
+def start_server(
```

Comment:
All of the setting up b2c stuff in here should go in a aiohttp startup function.